syntax = "proto3";

option go_package = "pb/game";

import "proto/common/tip.proto";
import "google/protobuf/empty.proto";
import "proto/logic/component/scene_comp.proto";
option cc_generic_services = true;


message GsEnterSceneRequest
{
  uint64 session_id = 1;
  uint64 scene_id = 2;
}


message GsLeaveSceneRequest
{
  bool change_gs = 1;
}


message EnterSceneS2CRequest
{
}

message EnterScenerS2CResponse
{
  TipInfoMessage error_message = 1;
}

service GamePlayerSceneService 
{
  rpc EnterScene (GsEnterSceneRequest) returns (google.protobuf.Empty);
  rpc LeaveScene (GsLeaveSceneRequest) returns (google.protobuf.Empty);
  rpc EnterSceneS2C (EnterSceneS2CRequest) returns (EnterScenerS2CResponse);
}